libs/coin-framework/src/account/helpers.ts

```diff
@@ -0,0 +1,414 @@
+  }
+};
+
+export const getAccountName = (account: AccountLike): string => {
+  switch (account.type) {
+    case "Account":
+    case "ChildAccount":
+      return account.name;
+
+    case "TokenAccount":
+      return account.token.name;
+
+    default:
+      throw new Error("invalid account.type=" + (account as AccountLike).type);
+  }
+};
+
+export const getAccountSpendableBalance = (account: AccountLike): BigNumber => {
+  switch (account.type) {
+    case "Account":
+    case "TokenAccount":
+      return account.spendableBalance;
+
+    case "ChildAccount":
+      return account.balance;
+
+    default:
+      throw new Error("invalid account.type=" + (account as AccountLike).type);
+  }
+};
+
+export const isAccountEmpty = (a: AccountLike): boolean => {
+  // if (a.type === "Account" && a.currency.family === "tron") {
+  //   const tronAcc = a as TronAccount;
+  //   // FIXME: here we compared a BigNumber to a number, would always return false
+  //   return (
+  //     tronAcc.tronResources && tronAcc.tronResources.bandwidth.freeLimit.eq(0)
+  //   );
+  // }
+
+  const hasSubAccounts =
+    a.type === "Account" && a.subAccounts && a.subAccounts.length;
+  return a.operationsCount === 0 && a.balance.isZero() && !hasSubAccounts;
+};
+
+export function areAllOperationsLoaded(account: AccountLike): boolean {
+  if (account.operationsCount !== account.operations.length) {
+    return false;
+  }
+
+  if (account.type === "Account" && account.subAccounts) {
+    return account.subAccounts.every(areAllOperationsLoaded);
+  }
+
+  return true;
+}
+
+export const isAccountBalanceSignificant = (a: AccountLike): boolean =>
+  a.balance.gt(100);
+
+// in future, could be a per currency thing
+// clear account to a bare minimal version that can be restored via sync
+// will preserve the balance to avoid user panic
+export function clearAccount<T extends AccountLike>(account: T): T {
+  if (account.type === "TokenAccount") {
+    return {
+      ...account,
+      balanceHistoryCache: emptyHistoryCache,
+      operations: [],
+      pendingOperations: [],
+    };
+  }
+
+  if (account.type === "ChildAccount") {
+    return {
+      ...account,
+      balanceHistoryCache: emptyHistoryCache,
+      operations: [],
+      pendingOperations: [],
+    };
+  }
+
+  const copy: Account = {
+    ...account,
+    balanceHistoryCache: emptyHistoryCache,
+    lastSyncDate: new Date(0),
+    operations: [],
+    pendingOperations: [],
+    subAccounts:
+      (account as Account).subAccounts &&
+      (account as Account).subAccounts?.map(clearAccount),
+  };
+
+  // if (copy.currency.family === "tron") {
+  //   const tronAcc = copy as TronAccount;
+  //   tronAcc.tronResources = {
+  //     ...tronAcc.tronResources,
+  //     cacheTransactionInfoById: {},
+  //   };
+  // }
+  // if (copy.currency.family === "bitcoin") {
+  //   (copy as BitcoinAccount).bitcoinResources = initialBitcoinResourcesValue;
+  // }
```

Comment:
we need a solution to allow the coins to define a clear cache. this is farely important, especially for the "clear" action to works on bitcoin family.